src/main/sensors/gyro.c

```diff
@@ -531,22 +538,20 @@ FAST_CODE void gyroFiltering(timeUs_t currentTimeUs)
     }
 #endif
 
-    if (!overflowDetected) {
-        for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-            gyroFilteredDownsampled[axis] = pt1FilterApply(&gyro.imuGyroFilter[axis], gyro.gyroADCf[axis]);
-        }
-    }
+    // Use clock cycle count at gyro interrupt to get as much precision as possible
```

Comment:
(gyro is internally using synchronous clock to generate samples. Using constant dt should be more accurate, if no samples are lost - it will remove readout jitter)

Reply:
Assuming a constant dt will effectively create a bias if that dt isn't exactly the average time between calculations. 
There is no derivation, only integration so noise caused by dividing by dt is not an issue.
What is important is that the dt reflects the actual time between measurements. I don't know a more accurate way of doing that than clock cycles between the interrupts from the IMU.

There are also significant jitter in when the attitude task runs, so the number of samples between calculations can vary.

And does all IMUs even have the option to be synchronized to the MCU clock?